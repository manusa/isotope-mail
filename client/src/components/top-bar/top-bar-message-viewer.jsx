import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import ButtonCollapse from './button-collapse';
import ButtonForward from './button-forward';
import ButtonMore from './button-more';
import ButtonReplyAll from './button-reply-all';
import {MessageViewerMenu} from './message-viewer-menu';
import TopBarButton from './top-bar-button';
import mainCss from '../../styles/main.scss';

export const TopBarMessageViewer = (
  {
    t, collapsed, sideBarToggle, clearSelectedMessage,
    outboxEmpty, onReplyAllMessageClick, onForwardMessageClick,
    onDeleteClick,
    onMarkUnreadClick
  }) => (
  <div className={mainCss['mdc-top-app-bar__row']}>
    <section className={`${mainCss['mdc-top-app-bar__section']} ${mainCss['mdc-top-app-bar__section--align-start']}`}>
      <ButtonCollapse collapsed={collapsed} sideBarToggle={sideBarToggle} />
      <span isotip={t('topBar.backToList')} isotip-position='bottom-start' isotip-size='small'>
        <TopBarButton
          className={`${mainCss['mdc-top-app-bar__navigation-icon']}`}
          onClick={clearSelectedMessage}>
          arrow_back
        </TopBarButton>
      </span>
    </section>
    <section className={`${mainCss['mdc-top-app-bar__section']} ${mainCss['mdc-top-app-bar__section--align-end']}`}>
      <ButtonReplyAll outboxEmpty={outboxEmpty} replyAllMessage={onReplyAllMessageClick}/>
      <ButtonForward outboxEmpty={outboxEmpty} forwardMessage={onForwardMessageClick}/>
      <span isotip={t('topBar.delete')} isotip-position='bottom' isotip-size='small'>
        <TopBarButton onClick={onDeleteClick}>delete</TopBarButton>
      </span>
      <span isotip={t('topBar.markUnread')} isotip-position='bottom-end' isotip-size='small'>
        <TopBarButton onClick={onMarkUnreadClick}>markunread</TopBarButton>
      </span>
      <ButtonMore><MessageViewerMenu /></ButtonMore>
    </section>
  </div>
);

TopBarMessageViewer.propTypes = {
  collapsed: PropTypes.bool.isRequired,
  sideBarToggle: PropTypes.func.isRequired,
  clearSelectedMessage: PropTypes.func.isRequired,
  outboxEmpty: PropTypes.bool.isRequired,
  onReplyAllMessageClick: PropTypes.func.isRequired,
  onForwardMessageClick: PropTypes.func.isRequired,
  onDeleteClick: PropTypes.func.isRequired,
  onMarkUnreadClick: PropTypes.func.isRequired
};

export default translate()(TopBarMessageViewer);
