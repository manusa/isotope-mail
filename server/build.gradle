buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
