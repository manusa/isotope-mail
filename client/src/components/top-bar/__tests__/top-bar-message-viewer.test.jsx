import React from 'react';
import {shallow} from 'enzyme/build/index';
import {TopBarMessageViewer} from '../top-bar-message-viewer';

describe('TopBarMessageViewer component test suite', () => {
  test('Snapshot render, should render top bar message viewer', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: true, sideBarToggle: jest.fn(), clearSelectedMessage: jest.fn(),
      onDeleteClick: jest.fn(), onForwardMessageClick: jest.fn(), onMarkUnreadClick: jest.fn(),
      onReplyAllMessageClick: jest.fn(), outboxEmpty: true};
    // When
    const topBarMessageViewer = shallow(<TopBarMessageViewer {...props}/>);
    // Then
    expect(topBarMessageViewer).toMatchSnapshot();
  });
});
