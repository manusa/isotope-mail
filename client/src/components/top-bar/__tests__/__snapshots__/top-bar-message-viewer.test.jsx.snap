// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarMessageViewer component test suite Snapshot render, should render top bar message viewer 1`] = `
<div
  className="mdc-top-app-bar__row"
>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
  >
    <Translate(ButtonCollapse)
      collapsed={true}
      sideBarToggle={[MockFunction]}
    />
    <span
      isotip="topBar.backToList"
      isotip-position="bottom-start"
      isotip-size="small"
    >
      <TopBarButton
        className="mdc-top-app-bar__navigation-icon"
        onClick={[MockFunction]}
      >
        arrow_back
      </TopBarButton>
    </span>
  </section>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
  >
    <Translate(ButtonReply)
      outboxEmpty={true}
      replyMessage={[MockFunction]}
    />
    <Translate(ButtonForward)
      forwardMessage={[MockFunction]}
      outboxEmpty={true}
    />
    <span
      isotip="topBar.delete"
      isotip-position="bottom"
      isotip-size="small"
    >
      <TopBarButton
        className=""
        onClick={[MockFunction]}
      >
        delete
      </TopBarButton>
    </span>
    <span
      isotip="topBar.markUnread"
      isotip-position="bottom-end"
      isotip-size="small"
    >
      <TopBarButton
        className=""
        onClick={[MockFunction]}
      >
        markunread
      </TopBarButton>
    </span>
    <Translate(ButtonMore)>
      <Connect(Translate(MessageViewerMenu)) />
    </Translate(ButtonMore)>
  </section>
</div>
`;
